### PR TITLE
skip this test until string guts are in our submission branch

### PR DESCRIPTION
**Pull request by lancep**

### CODE REVIEW

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
cc @parkera

File: TestFoundation/TestNSString.swift [hunk 1]
```diff
@@ -1211,22 +1211,41 @@ class TestNSString : XCTestCase {
 }
 
 struct ComparisonTest {
+    enum TestBehavior {
+      case run
+      case xfail(String)
```
Comment (phausler) [inline TestFoundation/TestNSString.swift]: xfail may be used in other places but is not very readable imho. Can we expand this to perhaps expectedFailure or something more indicative of what it is meant to be used for?

File: TestFoundation/TestNSString.swift [hunk 2]
```diff
@@ -1385,6 +1405,10 @@ func checkHasPrefixHasSuffix(_ lhs: String, _ rhs: String, _ stack: [UInt]) -> I
 extension TestNSString {
     func test_PrefixSuffix() {
         for test in comparisonTests {
+            if case .skip = test.behavior {
```
Comment (phausler) [inline TestFoundation/TestNSString.swift]: the PR is claiming a skip of a single test but that seems rather contrary to the additions and refactoring of reasons to xfail etc. Can we split this up into two parts?

File: TestFoundation/TestNSString.swift (hunk 1 above)
Comment (lancep) [reply TestFoundation/TestNSString.swift]: Sure, I'll make the change.

File: TestFoundation/TestNSString.swift (hunk 2 above)
Comment (lancep) [reply TestFoundation/TestNSString.swift]: Normally, I'd be fine with that but we're trying to unblock testing and I don't think it's worth while to have 2 PRs for such a small change. I can split it up if you insist but I'd rather just unblock CI.

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
(trying again because it looks like the timeouts are at least partially resolved)

@swift-ci please test

**Comment by alblue** [comment]
@swift-ci please clean test

**Comment by lancep** [comment]
@swift-ci please clean test

**Comment by lancep** [comment]
@swift-ci please clean test

**Comment by shahmishal** [comment]
@swift-ci please clean test

**Comment by shahmishal** [comment]
@swift-ci please test

**Comment by shahmishal** [comment]
@lancep "clean" is not supported with this repo.

**Comment by jrose-apple** [comment]
We really need to unblock this. Can we get a manager's permission to force-merge?

**Comment by lancep** [comment]
Mishal bumped the timeout, let's see if it passes this time

@swift-ci please test

**Comment by shahmishal** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test and merge

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by lancep** [comment]
@swift-ci Please test

**Comment by lancep** [comment]
@swift-ci Please test

**Comment by lancep** [comment]
@swift-ci Please test

**Pull request merged**